sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -8,12 +8,30 @@
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Azure.Core.Pipeline
 {
     internal static class TaskExtensions
     {
+        public static T WaitWithCancellation<T>(this Task<T> task, CancellationToken cancellationToken)
+        {
+            try
+            {
+                task.Wait(cancellationToken);
+            }
+            catch (Exception)
+            {
+                // ignore exception here to rethrow it later
+            }
+
+            return task.GetAwaiter().GetResult();
```

Comment:
This wouldn't this block even if cancellationToken is cancelled?

Reply:
Great catch! Fixed.